lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3606,7 +3741,21 @@ class VJPEmitter final
 
     // Get the parameter indices required for differentiating this function.
     SmallVector<SILValue, 4> allResults;
-    allResults.push_back(ai);
+    // Only append the results from the `destruct_tuple` instruction which are
+    // active, we don't consider the result of the original apply if it's a
+    // tuple.
+    if (ai->getResult(0)->getType().is<TupleType>()) {
+      auto *destructure =
+          getSingleUseOfTupleInDestructureTupleInst(ai->getResult(0));
+      if (destructure) {
+        for (auto result : destructure->getResults())
+          allResults.push_back(result);
+      } else {
+        allResults.push_back(ai);
+      }
+    } else {
+      allResults.push_back(ai);
```

Comment:
This has the duplicate line inside the if above when it's a tuple type, let me know if I should reformat this to avoid it, or if this is fine.

Reply:
Reformatted in https://github.com/apple/swift/pull/26743/commits/c5ebacc9df98d60f6f52d1c834d11c2a1b5a3620 by using helper boolean flags, e.g. [`bool isDifferentialAndFoundDestructureTupleUser`](https://github.com/apple/swift/pull/26743/commits/c5ebacc9df98d60f6f52d1c834d11c2a1b5a3620#diff-4ede198777b59748c377cdd17c444e83R1656).